3.preprocess-features/preprocess-features-utils.py

```diff
@@ -0,0 +1,129 @@
+from genericpath import isfile
+from importlib.resources import path
+import pathlib
+import pandas as pd
+import numpy as np
+import pigz_python
+import os
+
+from pycytominer.cyto_utils import DeepProfiler_processing
+from sklearn.preprocessing import StandardScaler
+
+def get_negative_control_index_df(original_index_df_path: pathlib.Path, annotations_path: pathlib.Path) -> pd.DataFrame:
+    """
+    create new df with only negative control images from original index df
+
+    Parameters
+    ----------
+    original_index_df_path : pathlib.Path
+        path to original index df used for DeepProfiler project
+    annotations_path : pathlib.Path
+        path to annotation metadata curated by IDR
+
+    Returns
+    -------
+    pd.DataFrame
+        new index df that only contains negative control images
+    """
+    original_index_df = pd.read_csv(original_index_df_path, dtype=str, index_col=0)
+    annotations = pd.read_csv(annotations_path)
+    
+    new_index_df = []
+    
+    # go through each row in orginal index df
+    for index, row in original_index_df.iterrows():
+        plate = row["Metadata_Plate"]
+        well = row["Metadata_Well"]
+        # if the image at row of original index df is a negative control, add it to new index df
+        image_annotations = annotations.loc[
+                    (plate == annotations["Plate"]) & (annotations["Well"] == well)
+                ]
+        control_type = image_annotations.iloc[0]["Control Type"]
+        if control_type == "negative":
+            new_index_df.append(row)
+        
+    return pd.DataFrame(new_index_df)
+
+def get_normalization_scaler(norm_pop_index_df_path: pathlib.Path, features_output_dir: pathlib.Path) -> StandardScaler:
```

Comment:
Is this somehow only selecting negative controls? I cannot tell

Reply:
The `pycytominer get_single_cells()` function compiles a pandas dataframe from an `index.csv` path. This function derives a scaler from whatever population is specified with the dataframe at `norm_pop_index_df_path`. Thus, when we pass a index df with only the negative controls, this function only selects negative controls to derive the normalization scaler from. Will make this more clear in the documentation